pkg/azurefile/nodeserver.go

```diff
@@ -142,6 +146,19 @@ func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRe
 	volumeID := req.GetVolumeId()
 	context := req.GetVolumeContext()
 	mountFlags := req.GetVolumeCapability().GetMount().GetMountFlags()
+	volumeMountGroup := req.GetVolumeCapability().GetMount().GetVolumeMountGroup()
+	gidExistsInMountFlags := false
+	if volumeMountGroup != "" {
+		for _, mountFlag := range mountFlags {
+			if strings.HasPrefix(mountFlag, "gid") {
+				gidExistsInMountFlags = true
+				kvpair := strings.Split(mountFlag, "=")
+				if !strings.EqualFold(volumeMountGroup, kvpair[1]) {
```

Comment:
I would suggest add this logic in a separate function, that would be easier to write ut.

Reply:
Done